src/polymer/polymer-element-mixin.ts

```diff
@@ -112,16 +112,14 @@ export class PolymerElementMixin extends ElementMixin implements
   }
 
   emitPropertyMetadata(property: PolymerProperty) {
-    const polymerMetadata:
-        {notify?: boolean, observer?: string, readOnly?: boolean} = {};
-    const polymerMetadataFields: Array<keyof typeof polymerMetadata> =
-        ['notify', 'observer', 'readOnly'];
-    for (const field of polymerMetadataFields) {
-      if (field in property) {
-        polymerMetadata[field] = property[field];
+    return {
+      polymer: {
+        notify: property.notify,
+        observer: property.observer,
+        readOnly: property.readOnly,
+        attributeType: property.attributeType,
       }
-    }
-    return {polymer: polymerMetadata};
+    };
```

Comment:
Ha… that's much simpler

Reply:
nice :)

Reply:
Yeah I was thinking maybe the original intention was to entirely omit fields whose value are undefined. But `JSON.stringify` does that automatically so it doesn't really matter, given that this is for the JSON format.

Reply:
That is the original intention. Provided that these don't include `null` then this will be fine.